infinispan-spring-boot-starter-remote/src/main/java/org/infinispan/spring/starter/remote/InfinispanJmxConfiguration.java

```diff
@@ -0,0 +1,31 @@
+package org.infinispan.spring.starter.remote;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jmx.export.MBeanExporter;
+
+/**
+ * Spring registers and Infinispan register the same bean. Avoid an exception telling Spring not to export Infinispan's
+ * bean
+ *
+ * @since 2.1.x
+ */
+@Configuration
+@ConditionalOnProperty(prefix = "infinispan.remote", name = "jmx", havingValue = "true")
+public class InfinispanJmxConfiguration {
+
+   private final ObjectProvider<MBeanExporter> mBeanExporter;
+
+   InfinispanJmxConfiguration(ObjectProvider<MBeanExporter> mBeanExporter) {
+      this.mBeanExporter = mBeanExporter;
+   }
+
+   @PostConstruct
+   public void validateMBeans() {
```

Comment:
Perhaps this should be named `addMBeanIfNecessary` or something like that?

Reply:
yes, makes much more sense